### ISSUE TITLE
[7.5] Table reverses rows when supplying initial sortBy state but without a column with corresponding accessor

### ISSUE THREAD

**Issue by theGirrafish** [tannerlinsley/react-table#2641]
**Describe the bug (required)**
The data has an additional numeric field (`myId`) that isn't inside the memoized column list, so it isn't being shown in the table.
The data is sorted beforehand by `myId` desc, so the rows with the largest `myId` should be on top.
Since it's initially sorted by `myId`, I set the initialState of the table to `{..., initialState: {sortBy: [{id: 'myId', desc: true}]}}`.

When the `myId` column doesn't exist, this causes the rows to be inverted when displayed on the table. This behavior is unexpected (at least afaik), especially since you can see in the console logs that the data array is properly sorted in desc order.

Also, when pressing `Show more` to expand the amount of data (I splice the data to show the first 5, then an extra 5, and so on), the additional data is added to the top instead of being inserted at the bottom of the table (I think because it reverse the supplied array).

The opposite happens when sorting the data asc beforehand, the table then displays everything in desc order and still adds to the top when showing more.

**TLDR:** The data array is being inverted when supplying an initialState sortBy where the id isn't in the list of supplied columns.

**Provide an example via Codesandbox! (required)**
I started my example based off the [existing sorting example](https://codesandbox.io/s/github/tannerlinsley/react-table/tree/master/examples/sorting).

https://codesandbox.io/s/nifty-wood-92rf0?file=/src/App.js

**Steps To Reproduce (required)**

1. Open the console and look at the order of the data by the `myId` key (it should be in desc order).
2. Look at the data in the table, the rows are sorted in asc order by `myId`
3. Click on the show more button. It should show 5 more rows, all added to the top, still in asc order by `myId`.
4. Open the console again and look at the order of the data by the `myId` key (it should still be in desc order - basically the data shown on the table but inverted).
5. To test the expected behavior, uncomment the column object containing the `myId` accessor
6. Everything should work normally, but with the `My Id` column in the table.

**Expected behavior (Recommended)**
The rows shouldn't be inverted before being displayed on the table when not supplying a column with the id that's used in the initial sortBy state.

**Desktop (please complete the following information):**
- chrome version 83

**Using React table version 7.5.0**

**Comment by tannerlinsley** [comment]
Thanks for noting this issue. To fix it, a breaking change is required and thus we will slate the fix for v8 and, for sake of triage, close this issue.

Thanks for your patience!
